__tests__/crowdfundings/cancelPledge.test.js

```diff
@@ -0,0 +1,161 @@
+const test = require('tape-async')
+const { apolloFetch, connectIfNeeded, pgDatabase } = require('../helpers.js')
+const { signIn, signOut, Users } = require('../auth.js')
+const { submitPledge } = require('./submitPledge.test.js')
+const { payPledge, PAYMENT_METHODS } = require('./payPledge.test.js')
+
+const CANCEL_PLEDGE_MUTATION = `
+  mutation cancelPledge($pledgeId: ID!) {
+    cancelPledge(pledgeId: $pledgeId) {
+      status
+      memberships {
+        active
+      }
+    }
+  }
+`
+
+const cancelPledge = async ({ pledgeId }) => {
+  await prepareParkingUserAndPledge()
+  return apolloFetch({
+    query: CANCEL_PLEDGE_MUTATION,
+    variables: {
+      pledgeId
+    }
+  })
+}
+
+module.exports = { cancelPledge }
+
+const prepareParkingUserAndPledge = async () => {
+  try {
+    await pgDatabase().public.pledges.insert({
+      id: process.env.PARKING_PLEDGE_ID,
+      userId: process.env.PARKING_USER_ID,
+      packageId: '00000000-0000-0000-0007-000000000001',
+      total: 0,
+      donation: 0
+    })
+  } catch (e) {
+    console.log(e)
+  }
+  try {
+    await pgDatabase().public.users.insert({
+      id: process.env.PARKING_USER_ID,
+      email: 'parking@test.project-r.construction',
+      firstName: 'parking',
+      lastName: 'parker'
+    })
+  } catch (e) {
+    console.log(e)
+  }
+}
+
+const prepareNewPledge = async ({ templateId } = {}) => {
+  const result = await submitPledge({
+    'total': 24000,
+    'options': [{
+      'amount': 1,
+      'price': 24000,
+      templateId: templateId || '00000000-0000-0000-0008-000000000001'
+    }]
+  })
+  return {
+    pledgeId: result.data.submitPledge.pledgeId,
+    userId: result.data.submitPledge.userId
+  }
+}
+
+const prepare = async (options) => {
+  await connectIfNeeded()
+  await pgDatabase().public.payments.truncate({ cascade: true })
+  await pgDatabase().public.pledgePayments.truncate({ cascade: true })
+  await pgDatabase().public.pledges.truncate({ cascade: true })
+  const newPledge = await prepareNewPledge(options)
+  return { ...newPledge }
+}
+
```

Comment:
do you want to check permissions centrally at some point? Otherwise you could check here if cancelPledge is only allowed for `supporter, admin`

Reply:
I'd like to postpone this as I have to come up with a general approach through helpers to test permissions first, if this is okay for you.